indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -256,8 +258,8 @@ public Response apply(TaskActionClient taskActionClient)
               retMap = Maps.newHashMap();
               retMap.put("result", ret);
             }
-            catch (IOException e) {
-              log.warn(e, "Failed to perform task action");
+            catch (Throwable e) {
```

Comment:
But really, this will catch things like OOME, which is probably not want to be caught.

Suggest checking for Exception, and throwing if is a fatal type.